File: SW_Expert_Academy/java/D3_5789_현주의_상자_바꾸기.java
import java.io.*;
import java.util.*;

class Solution {
    static int n, q, tcNum = 1;

    public static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        int tc = Integer.parseInt(br.readLine());

        for (int i = 1; i <= tc; i++) {
            solution();
        }

        bw.flush();
        bw.close();
    }

    private static void solution() throws IOException {
        StringTokenizer st = new StringTokenizer(br.readLine());

        n = Integer.parseInt(st.nextToken());
        q = Integer.parseInt(st.nextToken());

        int[] boxes = new int[n + 1];

        int l, r;

        for (int i = 1; i <= q; i++) {
            st = new StringTokenizer(br.readLine());
            l = Integer.parseInt(st.nextToken());
            r = Integer.parseInt(st.nextToken());

            for (int j = l; j <= r; j++) {
                boxes[j] = i;
            }
        }

        bw.write("#" + tcNum + " ");

        for (int i = 1; i < boxes.length; i++) {
            bw.write(boxes[i] + " ");
        }
        bw.write("\n");

        tcNum++;
    }
}

File: SW_Expert_Academy/cpp/D3_5789_현주의_상자_바꾸기.cpp
#include <iostream>
#include <vector>
using namespace std;

int n, q;
int tcNum = 1;

void solution() {
	cin >> n >> q;

	vector<int> boxes(n + 1, 0);
	
	int l, r;

	for (int i = 1; i <= q; i++) {
		cin >> l >> r;

		for (int j = l; j <= r; j++) {
			boxes[j] = i;
		}
	}

	cout << "#" << tcNum << " ";

	for (int i = 1; i < boxes.size(); i++) {
		cout << boxes[i] << " ";
	}
	cout << "\n";

	tcNum++;
}

int main() {
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	int tc;
	cin >> tc;

	for (int i = 1; i <= tc; i++) {
		solution();
	}

	return 0;
}